res/controllers/Denon-DN-S3700.qml

```diff
@@ -0,0 +1,14 @@
+import QtQml
+
+import "Mixxx"
+
+MixxxController {
+    id: controller
+
+    function init() {
+        console.error(controller.controllerId, controller.debugMode);
+    }
+    function shutdown() {
+        console.error(`Shutting down ${controller.controllerId} with debug mode ${controller.debugMode}`);
+    }
```

Comment:
These hooks will be invoked on component instantiation and destruction. Assuming this was inspired from the existing screen rendering, `init` and `shutdown` are called at different time.

- `init` is called once the controller has started. It is the direct equivalent to the same hook in Javascript. Unlike the `constructor` (equivalent to `Component.onCompleted`), it is invoked once the engine is running.
- `shutdown` is called when the engine is about to shutdown. This allows the controller to perform some animation or screen clean up, which is required for some devices (e.g restoring the splash screen, playing a shutdown animation, clearing the screen to a static colour, ...). In case of `Component.onDestruction`, as the name suggest, you don't have a component to render anymore. 

In case of "pure" controller (without UI rendering), I think it makes sense to ditch these function, but if we want to eventually merge the engines together, they will have to remain. 
I believe you might be able to replace them with signals instead tho (e.g `MixxxController::onStart`, `MixxxController::onShutdown`)